extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/ReactiveResteasyMpClientProcessor.java

```diff
@@ -146,6 +154,58 @@ void registerHeaderFactoryBeans(CombinedIndexBuildItem index,
         }
     }
 
+    @BuildStep
+    void registerProviders(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        // now we register all values of @RegisterProvider for constructor reflection
+        IndexView index = combinedIndexBuildItem.getIndex();
+        List<AnnotationInstance> allInstances = new ArrayList<>(index.getAnnotations(REGISTER_PROVIDER));
+        for (AnnotationInstance annotation : index.getAnnotations(REGISTER_PROVIDERS)) {
+            allInstances.addAll(Arrays.asList(annotation.value().asNestedArray()));
+        }
+        for (AnnotationInstance annotationInstance : allInstances) {
+            reflectiveClass
+                    .produce(new ReflectiveClassBuildItem(false, false, annotationInstance.value().asClass().toString()));
+        }
+
+        // Register @RegisterClientHeaders for reflection
+        for (AnnotationInstance annotationInstance : index.getAnnotations(REGISTER_CLIENT_HEADERS)) {
+            AnnotationValue value = annotationInstance.value();
+            if (value != null) {
+                reflectiveClass
+                        .produce(new ReflectiveClassBuildItem(false, false, annotationInstance.value().asClass().toString()));
+            }
+        }
+
+        // now retain all un-annotated implementations of ClientRequestFilter and ClientResponseFilter
+        // in case they are programmatically registered by applications
+        for (ClassInfo info : index.getAllKnownImplementors(CLIENT_REQUEST_FILTER)) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, info.name().toString()));
+        }
+        for (ClassInfo info : index.getAllKnownImplementors(CLIENT_RESPONSE_FILTER)) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, info.name().toString()));
+        }
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem registerProviderBeans(CombinedIndexBuildItem combinedIndex) {
```

Comment:
This was copied almost verbatim from the old rest-client

Reply:
ditto

Reply:
Same comment as above

Reply:
I tried it and without this, the filters I added in the test are **not**  CDI beans

Reply:
sorry, I misread the code, I thought this is also registering classes for reflection. This is the part that I think should not be necessary anymore

Reply:
👍🏼 

I removed that part